src/test/java/com/example/chapter1/basicprogramming/BinarySearchRecursivelyTest.java

```diff
@@ -0,0 +1,48 @@
+package com.example.chapter1.basicprogramming;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+public class BinarySearchRecursivelyTest {
+
+    @Test
+    public void searchAvailability() {
+        int[] input = new int[]{2, 4, 43, 44, 57};
```

Comment:
Allowing these duplicates for now as we are not going to have a member variable for it.